src/controllers/currency-conversion.controller.js

```diff
@@ -0,0 +1,108 @@
+import acceptedCurrencies from '../data/accepted-currencies.json';
+import log from '../services/logger.service';
+import { getExchangeRate } from '../services/quotation-of-the-day.service';
+
+const calculateExchange = (from, to, amount) => {
+    if (from <= 0 || to <= 0 || amount <= 0) {
+        log(3, 'currency-conversion.controller.js', 'calculateExchange()');
+        return {
+            error: true,
+            convertedValue: '0',
+        };
+    }
+
+    const amountUSD = (amount / from);
+    const amountTo = (amountUSD * to).toFixed(2);
+    return {
+        error: false,
+        convertedValue: amountTo,
+    };
+};
+
+const convert = async (req, res) => {
+    let { from, to, amount } = req.query;
+    const { accept } = acceptedCurrencies;
+
+    if (!from) {
+        const e = { error: true, message: 'Bad request: missing paramentreo FROM in query' };
+        log(3, 'currency-conversion.controller.js', `${e.message}`);
+        res.statusCode = 417;
+        res.send(e);
+        return false;
+    };
+
+    if (!to) {
+        const e = { error: true, message: 'Bad request: missing paramentreo TO in query' };
+        log(3, 'currency-conversion.controller.js', `${e.message}`);
+        res.statusCode = 417;
+        res.send(e);
+        return false;
+    };
+
+    from = from.toUpperCase();
```

Comment:
Consegue ver algum problema com o *re-assignment* de um argumento?

Reply:
Sim, entendo sua pergunta se considerarmos o paradigma da `programação funcional`, onde variáveis devem ser constantes e imutáveis. 
No entanto, nessa caso (e do `to`, na linha 51 também), optei em "reassinar" a variável uma vez, ao invés de chamar o `from.toUpperCase()` várias vezes (nas validações e conversão). Dito isso, acho que poderíamos criar uma const  `const fromUppercas = from.toUpperCase()` (mesma coisa para o `to`) e usar a mesma  nas validações e conversões.